### PR TITLE
chore: upgrade to polkadot-v0.9.24

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

## Changes

https://github.com/paritytech/substrate/compare/polkadot-v0.9.20...polkadot-v0.9.24

- https://github.com/paritytech/substrate/pull/10719
- https://github.com/paritytech/substrate/pull/8783
- https://github.com/open-web3-stack/open-runtime-module-library/pull/762
- https://github.com/open-web3-stack/open-runtime-module-library/pull/754
- https://github.com/open-web3-stack/open-runtime-module-library/pull/753

## Notes

- Always uses truncated account conversion (same behavior as on master)
- Migrated `jsonrpc` to `jsonrpsee`
- No use for `OnNewTokenAccount` && `OnKilledTokenAccount`
- Added `AssetAuthority` as pass-through to `EnsureRoot`
- `ParachainMinFee` always returns `Some(u128::MAX)`

### CODE REVIEW

**Comment by gregdhill** [comment]
Updated the XCM weights in `runtime-tests` based on similar updates to Acala. @sander2 if you have time I would appreciate some documentation on how to update these in the future. See my comment [here](https://github.com/interlay/interbtc/pull/646/commits/81ab8b4ac5522c17da3f021892d73500716d5aec#diff-2ccdf647cd2842f134e4e5a29382b591cf308772e6ad29492033a87f7b0b8d9cR155-R158) for example, I've added some rough calculations how the `xcm_fee` is calculated but I'm still not sure where `weight = 800_000_000` comes from.

File: crates/btc-relay/rpc/src/lib.rs [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
 use sp_api::ProvideRuntimeApi;
 use sp_blockchain::HeaderBackend;
 use sp_runtime::{generic::BlockId, traits::Block as BlockT, DispatchError};
 use std::sync::Arc;
 
-pub use self::gen_client::Client as RefundClient;
 pub use module_btc_relay_rpc_runtime_api::BtcRelayApi as BtcRelayRuntimeApi;
 
-#[rpc]
+#[rpc(client, server)]
```
Comment (sander2) [inline crates/btc-relay/rpc/src/lib.rs:16]: Can we make use of the client part in subxt?

File: parachain/runtime/interlay/src/xcm_config.rs [hunk 2]
```diff
@@ -299,10 +299,10 @@ parameter_types! {
 parameter_type_with_key! {
     // Only used for transferring parachain tokens to other parachains using DOT as fee currency. Currently we do not support this, hence return MAX.
     // See: https://github.com/open-web3-stack/open-runtime-module-library/blob/cadcc9fb10b8212f92668138fc8f83dc0c53acf5/xtokens/README.md#transfer-multiple-currencies
-    pub ParachainMinFee: |location: MultiLocation| -> u128 {
+    pub ParachainMinFee: |location: MultiLocation| -> Option<u128> {
         #[allow(clippy::match_ref_pats)] // false positive
         match (location.parents, location.first_interior()) {
-            _ => u128::MAX,
+            _ => Some(u128::MAX),
```
Comment (sander2) [inline parachain/runtime/interlay/src/xcm_config.rs:305]:
```suggestion
            _ => None,
```
(also in other runtimes)

**Review by sander2 — CHANGES_REQUESTED** [review]

File: crates/btc-relay/rpc/src/lib.rs (hunk 1 above)
Comment (gregdhill) [reply crates/btc-relay/rpc/src/lib.rs:16]: Hopefully, but I haven't tested this yet.

**Review by sander2 — APPROVED** [review]

**Pull request merged**